models/Market.ts

```diff
@@ -0,0 +1,34 @@
+import { FixedProductMarketMaker } from "@/queries/omen";
+import { fromHex } from "viem";
+
+export class MarketModel {
+  data: FixedProductMarketMaker;
+  closingDate: Date;
+  answer: number;
+  isClosed: boolean;
+
+  constructor(market: FixedProductMarketMaker) {
+    this.data = market;
+    this.closingDate = new Date(+market.openingTimestamp * 1000);
+    this.answer =
+      market?.question?.currentAnswer &&
+      fromHex(market.question.currentAnswer, "number");
+    this.isClosed = !!this.answer;
+  }
+
+  isWinner(outcomeIndex: number) {
+    return this.answer == outcomeIndex;
+  }
+
+  isLoser(outcomeIndex: number) {
+    return this.answer && this.answer != outcomeIndex;
+  }
```

Comment:
This can be derived from `isWinner`, so doesn't make much sense to have both. And they have different logic

Reply:
Not really as we have to check if the market is closed. 

For the winner we only have ot check if answer is equal to outcome index, as if answer is null (marked closed) it will be diff, so no need for that check.
I think this logic is needed, as otherwise will have to do this on the component level.

Reply:
I'll replace this.answer with this.isClosed